fetch_mail.py

```diff
@@ -0,0 +1,160 @@
+"""
+Method1
+-------
+    1. imaplib module
+        -> Establish a connection
+        -> Select the mailbox
+        -> Search for unseen mails
+        -> fetch their id
+        -> get raw data of each id
+        -> perform regex to find msg. patterns
+        -> save into a file
+        -> quit()
+"""
+
+import imaplib  # For reading messages
+import logging  # For catching logs
+import email
+import re
+import email.policy
+import socket  # Require for imaplib. connection
+
+
+class LPForum:
+    def __init__(self, email, password):
+        self.email_user = email
+        self.email_pass = password
+
+    def __str__(self):
+        return str(self.email_user)
+
+    def login(self):
+        """ Creating a secure connection over SSL socket
+        Return an imap4 object to "mail" variable """
+        try:
+            mail = imaplib.IMAP4_SSL("imap.gmail.com", 993)
+
+            # login to user account
+            mail.login(self.email_user, self.email_pass)
+            return mail
+        except imaplib.IMAP4.error as err:
+            logging.info(err)
+            return False
+
+    def get_message_id(self, obj):
+        """ Method for Return the number of mails
+        specified in the mailbox. """
+
+        obj.select(mailbox="INBOX")  # Select a mailbox first
+        _, msgnums = obj.search(
+            "utf-8", "(Unseen)", "FROM", "linkinpark@discoursemail.com"
```

Comment:
Having strings like this is known as "hard coding". Its OK to leave, but its nice if this is a variable set on the class so its easier to change in the future.

e.g. def __init__(self, email, password, subscribed_email)

then its easier to change.